scripts/meilisearchIndexer.ts

```diff
@@ -43,27 +68,46 @@ const updateMeilisearchIndexes = async () => {
         parentSection: frontMatter.parentSection,
         content: content.trim(),
         modified: convertMsToUnixSeconds(fileStats.mtimeMs),
-        group: frontMatter.objectID ? frontMatter.objectID.split('|')[0] : '',
+        group: frontMatter.objectID ? frontMatter.objectID.split('_')[0] : '',
         // viewed: 0 // this field will be added from UI upon screen load
       };
 
-      docs[f] = indexObj;
+      innerDocs[f] = indexObj;
     }
 
-    const res = await client.index(indexName).addDocuments(docs);
-    console.log(`Enqueued the ${res.type} for ${docs.length} records in Meilisearch with taskUid: ${res.taskUid}`);
+    allDocs.push(innerDocs);
+  }
 
-    let status = res.status;
+  // Write new entries
+  for (const innerDocs of allDocs) {
+    const res = await client.index(indexName).addDocuments(innerDocs);
+    console.log(`Enqueued the ${res.type} for ${innerDocs.length} records in Meilisearch with taskUid: ${res.taskUid}`);
 
-    while (status !== 'succeeded') {
-      new Promise(resolve => setTimeout(resolve, 1000));
-      console.log('Querying status...');
-      const taskInfo = await client.getTask(res.taskUid);
-      status = taskInfo.status;
-      if (status === 'failed') throw new Error(JSON.stringify(taskInfo));
-    }
+    await waitTxComplete(client, res.taskUid);
+    console.log(`${innerDocs.length} entries updated!`);
+  }
+
+  // Delete old entries
+  const toDelete = Object.keys(oldEntries);
+  if (toDelete.length) {
+    const deleteRes = await client.index(indexName).deleteDocuments({ filter: `objectID IN [${toDelete.join(',')}]` });
+    console.log(`Enqueued the ${deleteRes.type} of ${toDelete.length} records`);
+    waitTxComplete(client, deleteRes.taskUid);
+    console.log('Entries successfully deleted.');
+  }
+
+  console.log('\nSuccessfully updated all docs! :)');
+};
+
+const waitTxComplete = async (client: MeiliSearch, taskUid: number) => {
+  let status;
 
-    console.log('Index successfully updated');
+  while (status !== 'succeeded') {
+    new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
I don't get this one. If it's to set the polling interval, should probably be also `await` (or just use `setInterval` in the first place)

Reply:
Otherwise it seems it iterates again and again and again, as fast as js allows

Reply:
You are right, at least it was awaiting line 108, but yeah, it would immediately retry after receiving a response